conans/test/integration/lockfile/test_ci.py

```diff
@@ -0,0 +1,363 @@
+
+    # Do an unrelated change in A, should not be used, this is not the change we are testing
+    client.save({"pkga/myfile.txt": "ByeA World!!"})
+    client.run("create pkga pkgawin/0.2@ -s os=Windows")
+
+    # Do a change in B, this is the change that we want to test
+    client.save({"pkgb/myfile.txt": "ByeB World!!"})
+
+    # Test that pkgb/0.2 works
+    client.run("create pkgb pkgb/0.2@ -s os=Windows "
+               "--lockfile=app1.lock --lockfile-out=app1_b_changed.lock")
+    assert "pkgb/0.2: DEP FILE pkgawin: HelloA" in client.out
+
+    # Now lets build the application, to see everything ok
+    client.run("info app1/0.1@ --lockfile=app1_b_changed.lock --dry-build=missing "
+               "--build-order=build_order.json -s os=Windows")
+    json_file = client.load("build_order.json")
+    assert "app1/0.1" in json_file
+    assert "pkgc/0.1" in json_file
+    assert "pkga" not in json_file
+    assert "pkgb" not in json_file
+    to_build = json.loads(json_file)
+
+    for ref, _, _, _ in to_build:
+        client.run("install %s --build=%s --lockfile=app1_b_changed.lock  -s os=Windows"
+                   % (ref, ref))
+
+        assert "pkgawin/0.1:cf2e4ff978548fafd099ad838f9ecb8858bf25cb - Cache" in client.out
+        assert "pkgb/0.2:bf0518650d942fd1fad0c359bcba1d832682e64b - Cache" in client.out
+        assert "pkgb/0.2" in client.out
+        assert "pkgb/0.1" not in client.out
+        assert "DEP FILE pkgawin: HelloA" in client.out
+        assert "DEP FILE pkgb: ByeB World!!"
+
+
+@pytest.mark.xfail(reason="lockfiles wip")
+def test_multi_config_bundle(client_setup):
+    client = client_setup
+    client.run("lock create --reference=pkgd/0.1@ --lockfile-out=pkgd.lock "
+               "-s os=Windows ")
+    pkgd_lockfile = client.load("pkgd.lock")
+    print(pkgd_lockfile)
+
+    # Do an unrelated change in A, should not be used, this is not the change we are testing
+    client.save({"pkga/myfile.txt": "ByeA World!!"})
+    client.run("create pkga pkga/0.2@ -s os=Windows")
+
+    # Do a change in B, this is the change that we want to test
+    client.save({"pkgb/myfile.txt": "ByeB World!!"})
+
+    # Test that pkgb/0.2 works
+    client.run("create pkgb pkgb/0.2@ -s os=Windows "
+               "--lockfile=pkgd.lock --lockfile-out=b_win.lock")
+    assert "SELF OS: Windows!!" in client.out
+    assert "pkgb/0.2: DEP FILE pkga: HelloA" in client.out
+    b_win = client.load("b_win.lock")
+    print(b_win)
+    client.run("create pkgb pkgb/0.2@ -s os=Linux "
+               "--lockfile=pkgd.lock --lockfile-out=b_linux.lock")
+    assert "SELF OS: Linux!!" in client.out
+    assert "pkgb/0.2: DEP FILE pkga: HelloA" in client.out
+    b_linux = client.load("b_linux.lock")
+    print(b_linux)
+
+    # To bundle, lockfiles should be complete?
+    client.run("lock bundle create b_win.lock b_linux.lock")
+    print(client.load("lock.bundle"))
+    return
```

Comment:
What is the current status of the bundles with this implementation?

Reply:
Oh, I just read your comment `"bundle" operations are not implemented yet, the "multi-product" pipeline might still need it, but is not completely clear.`

Reply:
I would say the bundles are still needed right? otherwise, the packages not contained in 1 application will be clean from the second land the other way around right?
Should the bundle be just a merge of the lists? could/shold the isolated projects resolve to different versions of the dependencies? 
 I would talk about this to really understand.